Dockerfile

```diff
@@ -1,6 +1,10 @@
 FROM imandra/imandra-client-switch as fix-engine-build
 
 COPY --chown=opam:nogroup ./fix-engine.opam .
+COPY --chown=opam:nogroup ./vendor/imandra-ptime/imandra-ptime.opam .
+COPY --chown=opam:nogroup ./vendor/imandra-prelude/imandra-prelude.opam .
 RUN opam install . --deps-only --with-test --working-dir -y
+RUN rm imandra-ptime.opam imandra-prelude.opam
```

Comment:
The suggestion of 

```
COPY --chown=opam:nogroup ./fix-engine.opam .
COPY --chown=opam:nogroup ./vendor/imandra-ptime/imandra-ptime.opam vendor/imandra-ptime/imandra-ptime.opam
COPY --chown=opam:nogroup ./vendor/imandra-prelude/imandra-ptime.opam vendor/imandra-ptime/imandra-prelude.opam
```
along with:
```
opam install . \
   ./vendor/imandra-ptime/imandra-ptime.opam \
   ./vendor/imandra-ptime/imandra-prelude.opam \
  --deps-only --with-test --working-dir -y
```

would do away with the need for the `rm` here - as you say the later `COPY` step would then just override the vendor/*/*.opam files.

(You could add a `Make` step for this `opam install ... --deps-only ...` line too potentially, can COPY the `Makefile` over explicitly first too) - this tends to be what we do in other repos.

Reply:
Thanks. I'll take the last method.